### ISSUE TITLE
how to testing async program?

### ISSUE THREAD

**Issue by linying1991** [SketchUp/testup-2#197]
> The following content is translated by machine from DeepL, please forgive if the language is not correct

There are many asynchronous methods in SketchUp, such as UI::start_timer, HtmlDialog#add_action_callback, we sometimes encapsulate these asynchronous methods (the result of encapsulation is naturally asynchronous), how do we unit test these? Thanks!

**Comment by linying1991** [comment]
Indeed, I searched quite a few keywords and couldn't find a way to test asynchronously. I'll look for another way.
Thanks for the reply!

**Comment by thomthom** [comment]
I asked around internally, we use Squish for UI test automation.

**Comment by linying1991** [comment]
> The following content is translated by machine from DeepL, please forgive if the language is not correct

Great, I'll look into it, thanks a lot~
